### PR TITLE
MULE-15327: Avoid VoidMuleEvent when timeout in group expiry arises

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: core/src/main/java/org/mule/routing/correlation/EventCorrelator.java [hunk 1]
```diff
@@ -415,8 +418,17 @@ protected void handleGroupExpiry(EventGroup group) throws MessagingException
...
-            throw new CorrelationTimeoutException(CoreMessages.correlationTimedOut(group.getGroupId()),
-                                                  groupCollectionEvent);
+            
+            // This is intended to avoid an error in the handling of the correlation exception
+            // in case no event group was created before the timeout (no message for aggregate
+            // were received yet). A "custom" mule event is created so that a message is available.
+            if (groupCollectionEvent instanceof VoidMuleEvent)
+            {
+                groupCollectionEvent =
+                        new DefaultMuleEvent(new DefaultMuleMessage("No Group Collection Event was created before the timeout", muleContext), MessageExchangePattern.ONE_WAY, flowConstruct);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/routing/correlation/EventCorrelator.java]: use a NullPayload

Comment (elrodro83) [inline core/src/main/java/org/mule/routing/correlation/EventCorrelator.java]: static import

File: tests/integration/src/test/java/org/mule/routing/outbound/TestExceptionStrategy.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+package org.mule.routing.outbound;
+
+import org.mule.api.MuleEvent;
+import org.mule.api.exception.MessagingExceptionHandler;
+import org.mule.api.exception.RollbackSourceCallback;
+import org.mule.api.exception.SystemExceptionHandler;
+import org.mule.exception.AbstractExceptionListener;
+
+public class TestExceptionStrategy extends AbstractExceptionListener implements MessagingExceptionHandler, SystemExceptionHandler
+    {
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/routing/outbound/TestExceptionStrategy.java]: fix identation

File: core/src/main/java/org/mule/routing/correlation/EventCorrelator.java (hunk 1 above)
Comment (fsgonz) [reply core/src/main/java/org/mule/routing/correlation/EventCorrelator.java]: No static import to make it clear in NullPayload.getInstance().

**Pull request merged**